packages/client/src/lobby/createReplayJSON.ts

```diff
@@ -50,23 +49,51 @@ export default function createJSONFromReplay(room: string) {
   game.actions = getGameActionsFromState(replay);
 
   // Add the hypothesis from log, after current segment
-  const states = replay.hypothetical!.states;
-  const log = states[states.length - 1].log;
-  if (replay.segment < log.length) {
-    const slice = log.slice(replay.segment + 1);
-    const actions = getGameActionsFromLog(slice);
-    game.actions.push(...actions);
+  if (replay.hypothetical !== null) {
+    const states = replay.hypothetical.states;
+    const log = states[states.length - 1].log;
+    if (replay.segment < log.length) {
+      const slice = log.slice(replay.segment + 1);
+      const actions = getGameActionsFromLog(slice);
+      game.actions.push(...actions);
+    }
+  }
+
+  // Enforce at least one action
+  if (game.actions.length === 0) {
+    chat.addSelf(
+      '<span class="red">Error</span>: There are no actions in your hypo.',
+      room,
+    );
+    return;
   }
 
   const json = JSON.stringify(game);
-  navigator.clipboard.writeText(json).then(
-    () => {},
-    () => {},
-  );
-  chat.addSelf(
-    '<span class="green">Info</span>: Your hypo is copied on your clipboard.',
-    room,
-  );
+  const URLData = shrink(json);
+  if (URLData === "" || URLData === null) {
+    chat.addSelf(
+      '<span class="red">Error</span>: Your JSON data cannot be compressed.',
+      room,
+    );
+  } else {
+    const URL = `https://hanab.live/shared-replay-json/${URLData}`;
+    navigator.clipboard.writeText(URL).then(
+      () => {},
+      () => {},
+    );
+    chat.addSelf(
+      '<span class="green">Info</span>: Your hypo URL is copied on your clipboard.',
+      room,
+    );
+    const here = `<button href="#" onclick="navigator.clipboard.writeText('${json.replace(
```

Comment:
1) violates SLR
2) using .then for no reason
3) it should be </strong> instead of < /strong>
4) it should be </button> instead of </button >